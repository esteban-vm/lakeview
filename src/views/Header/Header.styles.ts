import clsx from 'clsx'

const styles = {
  wrapper: clsx('relative h-screen [&>img]:select-none [&>img]:object-cover'),
  overlay: clsx('absolute left-0 top-0 z-10 size-full bg-gradient-to-br from-primary-light/50 to-primary-dark/50'),
  slide1: clsx('animate-fade1'),
  slide2: clsx('animate-fade2'),
  slide3: clsx('animate-fade3'),
  slide4: clsx('animate-fade4'),
  content: clsx(
    'absolute left-1/2 top-1/2 z-20 flex -translate-x-1/2 -translate-y-1/2 flex-col',
    'max-w-2xl items-center justify-center gap-5 text-center text-white'
  ),
  logo: clsx('h-[30rem] w-auto select-none border border-white contrast-150'),
  heading: clsx('block text-9xl uppercase tracking-[0.2rem]'),
  subheading: clsx('block font-satisfy text-5xl'),
}

export default styles
