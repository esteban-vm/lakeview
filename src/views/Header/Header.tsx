import type { ImageProps } from 'next/image'
import Image from 'next/image'
import styles from './Header.styles'

const slideProps: Omit<ImageProps, 'src' | 'alt'> = {
  fill: true,
  quality: 100,
  sizes: '100vw',
}

const headerTitle = 'Lakeview header'

export default function Header() {
  return (
    <header className={styles.wrapper} aria-labelledby={headerTitle} id='header'>
      <div className={styles.overlay} />
      <Image alt='Slide 1' src='/images/header-image-1.jpg' className={styles.slide1} {...slideProps} />
      <Image alt='Slide 2' src='/images/header-image-2.jpg' className={styles.slide2} {...slideProps} />
      <Image alt='Slide 3' src='/images/header-image-3.jpg' className={styles.slide3} {...slideProps} />
      <Image alt='Slide 4' src='/images/header-image-4.jpg' className={styles.slide4} {...slideProps} />
      <div className={styles.content}>
        <Image
          alt='Lakeview logo'
          src='/images/lakeview-logo.png'
          quality={100}
          width={216}
          height={216}
          className={styles.logo}
        />
        <h1 className={styles.heading} id={headerTitle}>
          A Great Lifestyle
        </h1>
        <h2 className={styles.subheading}>In a Spectacular Setting</h2>
      </div>
    </header>
  )
}
