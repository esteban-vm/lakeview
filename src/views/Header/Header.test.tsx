import { render, cleanup, within, screen } from '@/tests'
import Header from './Header'

describe('Header:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Header />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', async () => {
    const headerElem = screen.getByRole('banner')
    expect(headerElem).toBeInTheDocument()
    expect(headerElem).toHaveAccessibleName(/a great lifestyle/i)
    expect(within(headerElem).getByRole('img', { name: /slide 1/i })).toBeInTheDocument()
    expect(within(headerElem).getByRole('img', { name: /slide 2/i })).toBeInTheDocument()
    expect(within(headerElem).getByRole('img', { name: /slide 3/i })).toBeInTheDocument()
    expect(within(headerElem).getByRole('img', { name: /slide 4/i })).toBeInTheDocument()
    expect(within(headerElem).getByRole('img', { name: /lakeview logo/i })).toBeInTheDocument()
    expect(within(headerElem).getByRole('heading', { name: /a great lifestyle/i, level: 1 })).toBeInTheDocument()
    expect(within(headerElem).getByRole('heading', { name: /in a spectacular setting/i, level: 2 })).toBeInTheDocument()
  })
})
