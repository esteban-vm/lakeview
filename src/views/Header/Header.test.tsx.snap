// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header: > should render correctly 1`] = `
<DocumentFragment>
  <header
    aria-labelledby="Lakeview header"
    class="relative h-screen [&>img]:select-none [&>img]:object-cover"
    id="header"
  >
    <div
      class="absolute left-0 top-0 z-10 size-full bg-gradient-to-br from-primary-light/50 to-primary-dark/50"
    />
    <img
      alt="Slide 1"
      class="animate-fade1"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=3840&q=100"
      srcset="/_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=640&q=100 640w, /_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=750&q=100 750w, /_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=828&q=100 828w, /_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimages%2Fheader-image-1.jpg&w=3840&q=100 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <img
      alt="Slide 2"
      class="animate-fade2"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=3840&q=100"
      srcset="/_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=640&q=100 640w, /_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=750&q=100 750w, /_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=828&q=100 828w, /_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimages%2Fheader-image-2.jpg&w=3840&q=100 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <img
      alt="Slide 3"
      class="animate-fade3"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=3840&q=100"
      srcset="/_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=640&q=100 640w, /_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=750&q=100 750w, /_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=828&q=100 828w, /_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimages%2Fheader-image-3.jpg&w=3840&q=100 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <img
      alt="Slide 4"
      class="animate-fade4"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=3840&q=100"
      srcset="/_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=640&q=100 640w, /_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=750&q=100 750w, /_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=828&q=100 828w, /_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimages%2Fheader-image-4.jpg&w=3840&q=100 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <div
      class="absolute left-1/2 top-1/2 z-20 flex -translate-x-1/2 -translate-y-1/2 flex-col max-w-2xl items-center justify-center gap-5 text-center text-color-white"
    >
      <img
        alt="Lakeview logo"
        class="h-[30rem] w-auto select-none border border-color-white contrast-150"
        data-nimg="1"
        decoding="async"
        height="216"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flakeview-logo.png&w=640&q=100"
        srcset="/_next/image?url=%2Fimages%2Flakeview-logo.png&w=256&q=100 1x, /_next/image?url=%2Fimages%2Flakeview-logo.png&w=640&q=100 2x"
        style="color: transparent;"
        width="216"
      />
      <h1
        class="block text-9xl uppercase tracking-[0.2rem]"
        id="Lakeview header"
      >
        A Great Lifestyle
      </h1>
      <h2
        class="block font-satisfy text-5xl"
      >
        In a Spectacular Setting
      </h2>
    </div>
  </header>
</DocumentFragment>
`;
